components/style/animation.rs

```diff
@@ -35,47 +158,68 @@ pub enum KeyframesIterationState {
     /// Infinite iterations, so no need to track a state.
     Infinite,
     /// Current and max iterations.
+    /// TODO: Everything else in this file uses f64, so perhaps this should
+    /// be as well.
     Finite(f32, f32),
 }
 
-/// This structure represents wether an animation is actually running.
-///
-/// An animation can be running, or paused at a given time.
-#[derive(Clone, Debug)]
-pub enum KeyframesRunningState {
-    /// This animation is paused. The inner field is the percentage of progress
-    /// when it was paused, from 0 to 1.
-    Paused(f64),
-    /// This animation is actually running.
-    Running,
-}
-
-/// This structure represents the current keyframe animation state, i.e., the
-/// duration, the current and maximum iteration count, and the state (either
-/// playing or paused).
-// TODO: unify the use of f32/f64 in this file.
+/// A CSS Animation
 #[derive(Clone)]
-pub struct KeyframesAnimationState {
+pub struct Animation {
+    /// The node associated with this animation.
+    pub node: OpaqueNode,
+
+    /// The name of this animation as defined by the style.
+    pub name: Atom,
+
+    /// The internal animation from the style system.
+    pub keyframes_animation: KeyframesAnimation,
+
     /// The time this animation started at.
     pub started_at: f64,
+
     /// The duration of this animation.
     pub duration: f64,
+
     /// The delay of the animation.
     pub delay: f64,
+
     /// The current iteration state for the animation.
     pub iteration_state: KeyframesIterationState,
-    /// Werther this animation is paused.
-    pub running_state: KeyframesRunningState,
+
+    /// Whether this animation is paused.
+    pub state: AnimationState,
+
     /// The declared animation direction of this animation.
     pub direction: AnimationDirection,
+
     /// The current animation direction. This can only be `normal` or `reverse`.
     pub current_direction: AnimationDirection,
+
     /// The original cascade style, needed to compute the generated keyframes of
     /// the animation.
     pub cascade_style: Arc<ComputedValues>,
+
+    /// Whether or not this animation is new and or has already been tracked
+    /// by the scrip thread.
+    pub is_new: bool,
```

Comment:
Would be it be useful to make this part of AnimationState instead?

Reply:
I originally used this design, but new animations are almost always handled like running animations. My hope is that in a future change, I would totally eliminate this field when adding support for better handling of `transition-delay` and `animation-delay`.